DataFormats/Detectors/ZDC/include/DataFormatsZDC/RawEventData.h

```diff
@@ -0,0 +1,146 @@
+// DataFormats/Detectors/ZDC/include/DataFormatsZDC/RawEventData.h
+
+#ifndef ALICEO2_ZDC_RAWEVENTDATA_H_
+#define ALICEO2_ZDC_RAWEVENTDATA_H_
+#include "CommonDataFormat/InteractionRecord.h"
+#include "CommonDataFormat/RangeReference.h"
+#include "ZDCBase/Constants.h"
+#include <Rtypes.h>
+#include <iostream>
+#include <gsl/span>
+
+// #include "Headers/RAWDataHeader.h"
+// #include <Framework/Logger.h>
+// #include <utility>
+// #include <cstring>
+
+/// \file BCData.h
+/// \brief Container of ZDC raw data
+/// \author pietro.cortese@cern.ch
+
+namespace o2
+{
+namespace zdc
+{
+
+constexpr unsigned short id_w0 = 0x0;
+constexpr unsigned short id_w1 = 0x1;
+constexpr unsigned short id_w2 = 0x2;
+constexpr int NWPerGBTW = 4;
+
+struct __attribute__((__packed__)) ChannelDataV0 {
+  // First GBT word
+  unsigned fixed_0 : 2;
+  unsigned board : 4;
+  unsigned ch : 2;
+  unsigned offset : 16;
+  unsigned hits : 12;
+  unsigned bc : 12;
+  unsigned orbit : 32;
+  unsigned empty_0 : 16;
+  unsigned empty_1 : 32;
+
+  // Second GBT word
+  unsigned fixed_1 : 2;
+  unsigned error : 2;
+  unsigned Alice_0 : 1;
+  unsigned Alice_1 : 1;
+  unsigned Alice_2 : 1;
+  unsigned Alice_3 : 1;
+  unsigned s00 : 12;
+  unsigned s01 : 12;
+  unsigned s02 : 12;
+  unsigned s03 : 12;
+  unsigned s04 : 12;
+  unsigned s05 : 12;
+  unsigned empty_2 : 16;
+  unsigned empty_3 : 32;
+
+  // Third GBT word
+  unsigned fixed_2 : 2;
+  unsigned Hit : 1;
+  unsigned Auto_m : 1;
+  unsigned Auto_0 : 1;
+  unsigned Auto_1 : 1;
+  unsigned Auto_2 : 1;
+  unsigned Auto_3 : 1;
+  unsigned s06 : 12;
+  unsigned s07 : 12;
+  unsigned s08 : 12;
+  unsigned s09 : 12;
+  unsigned s10 : 12;
+  unsigned s11 : 12;
+  unsigned empty_4 : 16;
+  unsigned empty_5 : 32;
+};
+
+struct __attribute__((__packed__)) ChannelDataV0W0 {
+  unsigned fixed_0 : 2;
+  unsigned board : 4;
+  unsigned ch : 2;
+  unsigned offset : 16;
+  unsigned hits : 12;
+  unsigned bc : 12;
+  unsigned orbit : 32;
+  unsigned empty_0 : 16;
+  unsigned empty_1 : 32;
+};
+
+struct __attribute__((__packed__)) ChannelDataV0W1 {
+  unsigned fixed_1 : 2;
+  unsigned error : 2;
+  unsigned Alice_0 : 1;
+  unsigned Alice_1 : 1;
+  unsigned Alice_2 : 1;
+  unsigned Alice_3 : 1;
+  unsigned s00 : 12;
+  unsigned s01 : 12;
+  unsigned s02 : 12;
+  unsigned s03 : 12;
+  unsigned s04 : 12;
+  unsigned s05 : 12;
+  unsigned empty_2 : 16;
+  unsigned empty_3 : 32;
+};
+
+struct __attribute__((__packed__)) ChannelDataV0W2 {
+  unsigned fixed_2 : 2;
+  unsigned Hit : 1;
+  unsigned Auto_m : 1;
+  unsigned Auto_0 : 1;
+  unsigned Auto_1 : 1;
+  unsigned Auto_2 : 1;
+  unsigned Auto_3 : 1;
+  unsigned s06 : 12;
+  unsigned s07 : 12;
+  unsigned s08 : 12;
+  unsigned s09 : 12;
+  unsigned s10 : 12;
+  unsigned s11 : 12;
+  unsigned empty_4 : 16;
+  unsigned empty_5 : 32;
+};
+
+struct EventData {
+  union {
+    UInt_t w[o2::zdc::NWPerBc][NWPerGBTW];
+    struct ChannelDataV0 f;
+  } data[o2::zdc::NModules][o2::zdc::NChPerModule] = {0};
```

Comment:
Did you check that this initialization to 0 works? An alternative way would be to define EventData as here: https://github.com/AliceO2Group/AliceO2/blob/14851f87b3a8bc708fb0742cf9ed2d4c2f148f7a/DataFormats/Headers/include/Headers/RAWDataHeader.h#L69-L131 (with the drawback that the root cannot generate streamers for anonymous structs, but this is not needed for the raw data).

Reply:
It seems to contain zeros, whatever the initializer in curly brackets. However the output structure needs to be initialized at every bunch crossing in Digits2Raw.cxx. Probably for the moment it is not needed to change the structure.